### DISCUSSION TITLE
Savegame compatibility?

### DISCUSSION THREAD

**Discussion by pzychotic** [DescentDevelopers/Descent3#366] [Q&A]
Working on a Windows x64 build ([#354](https://github.com/DescentDevelopers/Descent3/pull/354)), I noticed that I can't load my old savegames from the 32bit build.

Turns out, lots of in-game structs are just copied into the savegame as-is with `cf_WriteBytes()` and read back with` cf_ReadBytes()`.
Some of the involved structs could be made compatible with a `#pragma pack`. But others like `struct object` contain pointers which obviously can't.

I guess Linux/Mac has the same problem also.

Is this topic of a concern that should be looked into or is this a known limitation that just is how it is?

**Comment by GravisZro** [comment]
It seems like something that can be fixed, so why not fix it? All we need is an accurate map of the old structures to import the data. What should be done though is make a stable endian independent save game format. I've been looking into this problem.
